### DISCUSSION TITLE
Exporting Marimo notebook in code view

### DISCUSSION THREAD

**Discussion by smolinad** [marimo-team/marimo#432] [Q&A] [answered]
Hello everyone, 

I'd like to know whether a notebook can be exported from the code view instead of the app view. I'd like to export the cells showing the code, and not only the outputs and visualizations. 

Thanks in advance!

**Comment by mscolnick** [comment, marked as answer]
Hey @smolinad, we added this in [Release 0.1.64](https://github.com/marimo-team/marimo/releases/tag/0.1.64) via PR https://github.com/marimo-team/marimo/pull/440. 

Can you upgrade to the latest version and re-export your notebook - you should see the "code toggle" in the bottom left corner. 

Let us know if this fits your needs, or if you are looking for something more.

**Comment by eeshsaxena** [comment]
Yes, Marimo supports exporting notebooks in script form which preserves the code cells.

**Export as a Python script (shows all code):**

From the command line:

```bash
marimo export script your_notebook.py -o output_script.py
```

This produces a plain `.py` file with all your cell code preserved, not just the outputs.

**Export as HTML with code visible:**

```bash
marimo export html your_notebook.py -o output.html
```

By default the HTML export shows both the code and the outputs. You can also open the exported HTML and toggle the code view in the UI.

**From inside Marimo UI:**

Click the three-dot menu (top right) > Download > choose "Script (.py)" to get the full source with all cells intact.

The `.py` format is especially useful if you want to share the notebook as readable code or commit it to a repo, since Marimo notebooks are just Python files with `@app.cell` decorators. Opening the `.py` file directly in any editor shows all the code cleanly.
